### PR TITLE
Use network pressures from restart

### PR DESCRIPTION
**Pull request by vkip**

### CODE REVIEW

**Comment by vkip** [comment]
jenkins build this opm-common=3738 please

**Comment by vkip** [comment]
Tests depend on re-routing (https://github.com/OPM/opm-common/pull/3695 and https://github.com/OPM/opm-simulators/pull/4945), so those should go first.

File: opm/simulators/wells/BlackoilWellModelGeneric.cpp [hunk 1]
```diff
@@ -91,20 +91,27 @@ BlackoilWellModelGeneric(Schedule& schedule,
...
+                                          this->parallel_well_info_.end(),
+                                          value);
+
+        return (candidate == this->parallel_well_info_.end())
+                || (*candidate != value);
+    };
+
+    const auto& node_pressures = eclState.getRestartNetworkPressures();
+    if (node_pressures.has_value()) {
+        for (const auto& it: node_pressures.value()) {
```
Comment (GitPaean) [inline opm/simulators/wells/BlackoilWellModelGeneric.cpp:111]: Can we do 
```c++
this->node_pressures_ = node_pressures.value();
```?

Comment (vkip) [reply opm/simulators/wells/BlackoilWellModelGeneric.cpp:111]: Absolutely, that's better - will update. Thanks!

Comment (GitPaean) [reply opm/simulators/wells/BlackoilWellModelGeneric.cpp]: It is more like a question.  If there are more complicated scenarios that we can not do assigning, we will know and deal with it.

**Comment by vkip** [comment]
jenkins build this opm-common=3738 please

**Comment by vkip** [comment]
jenkins build this opm-common=3738 opm-tests=1090 please

**Comment by GitPaean** [comment]
Hei, do you have some understanding of the following comparison failures with 100% relative error?
```
	WGPGR:PROD3: Summary file
		Fails for 1 entries
		Largest absolute error: 4.5201600e+05
		Largest relative error: -1.0000000e+00
	WOPGR:PROD3: Summary file
		Fails for 1 entries
		Largest absolute error: 4.1408140e+03
		Largest relative error: -1.0000000e+00
```

**Comment by vkip** [comment]
> Hei, do you have some understanding of the following comparison failures with 100% relative error?
> 
> ```
> 	WGPGR:PROD3: Summary file
> 		Fails for 1 entries
> 		Largest absolute error: 4.5201600e+05
> 		Largest relative error: -1.0000000e+00
> 	WOPGR:PROD3: Summary file
> 		Fails for 1 entries
> 		Largest absolute error: 4.1408140e+03
> 		Largest relative error: -1.0000000e+00
> ```

Having problems reproducing this locally. Will trigger a rebuild to check if it could be some build system hiccup..

**Comment by vkip** [comment]
jenkins build this opm-common=3738 opm-tests=1090 please

**Comment by GitPaean** [comment]
FYI, the PR for opm-tests have been merged.

**Comment by vkip** [comment]
Ah.. see now it is the guide rate summary output vectors, where the restarted simulation outputs zero values. Nothing to do with this PR, probably best to remove these from the test? Will have a look at the other vectors with minor differences as well - suppose we might have to adjust tolerances a bit.

**Comment by GitPaean** [comment]
> Will have a look at the other vectors with minor differences as well - suppose we might have to adjust tolerances a bit.

If the well curve comparison looks good, then it is safe.

**Comment by vkip** [comment]
jenkins build this opm-common=3738 opm-tests=1092 please

**Comment by vkip** [comment]
jenkins build this opm-common=3738 opm-tests=1092 please

**Review by bska — APPROVED** [review]
Thanks a lot.  This looks good to me and I'll merge into master.

**Pull request merged**